Lib/contextlib.py

```diff
@@ -31,6 +32,28 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class AbstractAsyncContextManager(abc.ABC):
+
+    """An abstract base class for asynchronous context managers."""
+
+    async def __aenter__(self):
+        """Return `self` upon entering the runtime context."""
+        return self
+
+    @abc.abstractmethod
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        """Raise any exception triggered within the runtime context."""
+        return None
+
+    @classmethod
+    def __subclasshook__(cls, C):
+        if cls is AbstractAsyncContextManager:
+            if (any("__aenter__" in B.__dict__ for B in C.__mro__) and
+                any("__aexit__" in B.__dict__ for B in C.__mro__)):
```

Comment:
Maybe that should be a separate issue? `AbstractContextManager` should arguably be fixed even in 3.6, and this PR only targets 3.7. I can fix `AbstractAsyncContextManager` though.